### DISCUSSION TITLE
Internal SSL certificate + cloudflare tunnel

### DISCUSSION THREAD

**Discussion by emilinsweden** [w35l3y/hassio-addons#11] [Q&A]
Hello I really need help, I followed this tutorial https://hodgkins.io/securing-home-assitant-with-cloudflare. But I want to not have to port forward in my router. So I followed your tutorial and removed port forwarding but it is not working, how do I combine these? Or do I need to do something else?
 

This error I get
2022-01-29T17:27:08Z ERR  error="Unable to reach the origin service. The service may be down or it may not be responding to traffic from cloudflared: EOF" cfRay=6d542b664dae21c3-DUS ingressRule=0 originService=http://192.168.1.220:443

**Comment by w35l3y** [comment]
Here is the tutorial to use this add-on:
https://github.com/w35l3y/hassio-addons/tree/main/cloudflare_tunnel#readme

This already is protected with SSL.
You will only need this tutorial.

**Reply by emilinsweden** [reply]
Oh okay. Thank you, I removed the other one and it works
